frontend/model/contracts/group.js

```diff
@@ -1360,13 +1360,81 @@ sbp('chelonia/defineContract', {
-      process ({ meta }, { state, getters }) {
-        const period = getters.periodStampGivenDate(meta.createdDate)
+      process ({ meta, contractID }, { state, getters }) {
+        const periodTo = getters.periodStampGivenDate(meta.createdDate)
+        const periodLength = getters.groupSettings.distributionPeriodLength
         const current = state.settings?.distributionDate
-        if (current !== period) {
-          // right before updating to the new distribution period, make sure to update various payment-related group streaks.
-          updateGroupStreaks({ state, getters })
-          state.settings.distributionDate = period
+        const currentPeriodPayments = cloneDeep(getters.groupPeriodPayments[current])
+
+        if (comparePeriodStamps(periodTo, current) > 0) {
+          const getAllEmptyPeriodsBetween = () => {
+            const periods = []
+            let recorded = current
+            while (true) {
+              const period = plusOnePeriodLength(recorded, periodLength)
+              if (comparePeriodStamps(periodTo, period) > 0) {
+                periods.push(period)
+                recorded = period
+              } else {
+                break
+              }
+            }
+            return periods
+          }
+
+          // There can be one or more distribution periods without any payments (empty-periods) between current 'distributionDate'
+          // and the distribution period to update to. (eg. No one in the group hasn't logged in for a long time)
+          // 'paymentsByPeriod' object in the group state and the payment-related group streaks need to be populated/updated accordingly for those empty periods in this case.
+          // (Reference issue: https://github.com/okTurtles/group-income/issues/2982)
+          const emptyPeriodsBetweenCurrentAndTo = getAllEmptyPeriodsBetween()
+          const allPeriodsToUpdate = [
+            ...emptyPeriodsBetweenCurrentAndTo,
+            periodTo
+          ]
+
+          allPeriodsToUpdate.forEach((period, index) => {
+            initFetchPeriodPayments({ contractID, meta, periodTo: period, state, getters })
+          })
+
+          // There can be two cases:
+          // 1) The period to update comes right after current period
+          // 2) There are one or more empty periods (periods with no payments made) between current period and the period to update.
+          // In case of 1), we add +1 to the payment-related group streaks.
+          // In case of 2), those streaks need to be reset except for 'missedPayments' streak.
+          if (emptyPeriodsBetweenCurrentAndTo.length === 0) {
+            updateGroupStreaks({ state, getters })
+          } else {
+            state.streaks = {
+              ...state.streaks,
+              ...({
+                lastStreakPeriod: null,
+                fullMonthlyPledges: 0,
+                fullMonthlySupport: 0,
+                onTimePayments: {}
+              })
+            }
```

Comment:
This could be simplified to 4 lines instead of 9:

```js
            state.streaks.lastStreakPeriod = null
            state.streaks.fullMonthlyPledges = 0
            state.streaks.fullMonthlySupport = 0
            state.streaks.onTimePayments = {}
```

Reply:
Because `missedPayments` streaks still need to be increased(`+ number of empty periods`) instead of being reset.